### PR TITLE
fix(ProgressIndicator): refactor to not use incomplete prop

### PR DESCRIPTION
**Pull request by hellobrian**
Closes carbon-design-system/carbon-components-react#289

Refactor `ProgressIndicator` and `ProgressStep` so that it doesn't need to use `incomplete` prop.
Should now rely on `complete` prop being `true` or `false`. 

#### Changelog

**Changed**

- `ProgressIndicator` to work with `incomplete` prop

**Removed**

- code related to `incomplete` prop

### CODE REVIEW

File: components/__tests__/ProgressIndicator-test.js [hunk 1]
```diff
@@ -59,26 +59,38 @@ describe('ProgressIndicator', () => {
...
 
       it('should render with a label', () => {
-        expect(mountedList.find(ProgressStep).at(0).props().label).toEqual(
-          'label'
-        );
+        expect(
+          mountedList
+            .find(ProgressStep)
+            .at(0)
+            .props().label
```
Comment (joshblack) [inline components/__tests__/ProgressIndicator-test.js]: Could also be:

```js
.props('label')
```

I think 😄

File: components/__tests__/ProgressIndicator-test.js [hunk 2]
```diff
@@ -59,26 +59,38 @@ describe('ProgressIndicator', () => {
...
+        ).toEqual('label');
       });
 
       it('should render with a description', () => {
         expect(
-          mountedList.find(ProgressStep).at(0).props().description
+          mountedList
+            .find(ProgressStep)
+            .at(0)
+            .props().description
```
Comment (joshblack) [inline components/__tests__/ProgressIndicator-test.js]: Same as above

File: components/__tests__/ProgressIndicator-test.js [hunk 3]
```diff
@@ -92,7 +104,12 @@ describe('ProgressIndicator', () => {
         });
 
         it('should render a current ProgressStep with correct props', () => {
-          expect(list.find(ProgressStep).at(3).props().current).toBe(true);
+          expect(
+            list
+              .find(ProgressStep)
+              .at(3)
+              .props().current
```
Comment (joshblack) [inline components/__tests__/ProgressIndicator-test.js]: Same as above

File: components/__tests__/ProgressIndicator-test.js (hunk 1 above)
Comment (hellobrian) [reply components/__tests__/ProgressIndicator-test.js]: I think it's supposed to be `.props().label`, here are the [enzyme docs](https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/props.md)

Comment (joshblack) [reply components/__tests__/ProgressIndicator-test.js]: Sorry, I mistyped 😅 

http://airbnb.io/enzyme/docs/api/ShallowWrapper/prop.html

Comment (joshblack) [reply components/__tests__/ProgressIndicator-test.js]: e.g.

```diff
it('should render with a label', () => {
  expect(
    mountedList
      .find(ProgressStep)
      .at(0)
-     .props('label')
+     .prop('label')
  ).toEqual('label');
});
```

**Comment by marijohannessen** [comment]
@hellobrian can you update this when you have the chance :-)

File: components/__tests__/ProgressIndicator-test.js (hunk 1 above)
Comment (hellobrian) [reply components/__tests__/ProgressIndicator-test.js]: Sorry, this took a while to update, made the changes in the latest commit.

**Comment by hellobrian** [comment]
I see the merge conflicts, I'll resolve again this week

**Comment by joshblack** [comment]
@hellobrian let me know if you want me to do it as well 😄  don't want to take up too much of your time!

**Comment by hellobrian** [comment]
Didn't realize there was a lint error, fixed now @joshblack

**Comment by tw15egan** [comment]
I'm seeing some inconsistencies styling wise between the vanilla component + the react component after these changes. Here is the Vanilla component: 

screen shot 2017-11-15 at 10 49 34 am

And here is the React component: 

screen shot 2017-11-15 at 10 49 21 am


Seems like:
- [ ] Completed circle is completely filled, should have gape between border and inner circle
- [ ] The line extending left from the completed circle should be `$brand-01`

**Comment by hellobrian** [comment]
Nooo000oooo 😭 lol, I'll take another look at this. Thanks @tw15egan

**Comment by tw15egan** [comment]
lol sorry @hellobrian! No rush, we appreciate you taking time out to do this for us!! 😄

**Pull request merged**